jetty-websocket/javax-websocket-client-impl/src/main/java/org/eclipse/jetty/websocket/jsr356/JsrSessionTracker.java

```diff
@@ -49,7 +51,7 @@ public void onSessionClosed(JsrSession session)
     @Override
     protected void doStop() throws Exception
     {
-        for (JsrSession session : sessions)
+        for (Session session : sessions)
```

Comment:
while iterating over the set is thread safe, it doesn't prevent other sessions from being added while you are iterating.  So either you need to ensure that once you are stopping, new sessions can't be added, or you need some outer `while (sessions.size()>0)` loop.

Reply:
@gregw This is a pre-existing issue and not one that was created by this PR, so maybe better for a different PR. I have reopened #4919 for this issue as well.

What do you think of an approach like this:
https://gist.github.com/lachlan-roberts/6c52d13874628dd3ac5b5ecaaba91123

Reply:
That approach kind of invalidates this PR which is using a concurrent set and no locking, so really need an approach that works with concurrency... hence I think that it should be done as part of this PR.

Reply:
I think that checking `isRunning()` before adding is OKish.... but a little racy... which might be able to be handled by again checking isRunning after adding and closing the session just added?

Reply:
@gregw I think @lachlan-roberts has a point. 
If we will need to provide a patch to the sponsor, I would rather use the small patch that this PR proposes, rather than a larger patch also encompassing graceful shutdown which is orthogonal to this PR.

Reply:
OK - but let's not break the sponsor again with the other PR because the approach suggested was full of locks that could cause the problem again.